x-pack/plugins/transform/public/app/sections/create_transform/components/advanced_runtime_mappings_editor_switch/switch_modal.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC } from 'react';
+import { EuiConfirmModal } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+interface Props {
+  onCancel: () => void;
+  onConfirm: () => void;
+}
+
+const modalTitle = i18n.translate('xpack.transform.stepDefineForm.runtimeEditorSwitchModalTitle', {
+  defaultMessage: 'Edits will be lost',
+});
+
+const cancelButtonText = i18n.translate(
+  'xpack.transform.stepDefineForm.runtimeEditorSwitchModalCancelButtonText',
+  {
+    defaultMessage: 'Cancel',
+  }
+);
+
+const applyChangesText = i18n.translate(
+  'xpack.transform.stepDefineForm.runtimeEditorSwitchModalConfirmButtonText',
+  {
+    defaultMessage: 'Disable editor',
+  }
+);
+const modalMessage = i18n.translate(
+  'xpack.transform.stepDefineForm.runtimeEditorSwitchModalBodyText',
+  {
+    defaultMessage: `The changes in the advanced editor haven't been applied yet. By disabling the editor you will lose your edits.`,
```

Comment:
Switch `disabling` to `closing` if you prefer `Close editor`.

Reply:
Updated here [`2e96863` (#97008)](https://github.com/elastic/kibana/pull/97008/commits/2e968634beeb83e156b85f81d87115638fbec52b)